src/activity_item_handling.cpp

```diff
@@ -1392,6 +1392,32 @@ static cata::optional<tripoint> find_best_fire(
     return best_fire;
 }
 
+static cata::optional<tripoint> find_best_refuel_spot( const tripoint &center )
+{
+    const zone_manager &mgr = zone_manager::get_manager();
+    const zone_type_id zone_loot_wood( "LOOT_WOOD" );
+    const tripoint center_abs = g->m.getabs( center );
+
+    const std::unordered_set<tripoint> &tiles_abs_unordered =
+            mgr.get_near( zone_loot_wood, center_abs, PICKUP_RANGE );
+    const std::vector<tripoint> &tiles_abs =
+            get_sorted_tiles_by_distance( center_abs, tiles_abs_unordered );
+
+    cata::optional<tripoint> best_tile;
+
+    for( const tripoint &tile_abs : tiles_abs ) {
+        const tripoint tile = g->m.getlocal( tile_abs );
+        if( g->m.has_items( tile ) &&
+                g->m.accessible_items( tile ) &&
+                g->m.clear_path( center, tile, PICKUP_RANGE, 1, 100 ) ) {
+            best_tile = tile;
```

Comment:
You can return directly from here.

Btw. `best_tile` is not really fitting anyway as it is not comparing different suitable tiles, it only returns the first encountered suitable tile.

Reply:
Indeed, you are right.

But, excuse my C++ newbe ignorance, but then how do I return en empty `cata::optional<tripoint>` at the end of the function without declaring a variable of that type first ?

Reply:
CLIon is suggesting me to simply `return cata::optional<tripoint>();`

Reply:
Or simply `return {};`, which returns a default-constructed object.